fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AwesomeBarState.kt

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.appstate
+
+import mozilla.components.concept.awesomebar.AwesomeBar
+import mozilla.components.feature.fxsuggest.FxSuggestInteractionInfo
+import mozilla.components.feature.fxsuggest.facts.FxSuggestFacts
+import org.mozilla.fenix.GleanMetrics.FxSuggest
+import org.mozilla.fenix.search.SearchDialogFragment
+
+/**
+ * State for interactions with the [AwesomeBar] in the [SearchDialogFragment].
+ *
+ * @property visibilityState The suggestions and groups that are currently displayed in the [AwesomeBar].
+ * @property clickedSuggestion The [AwesomeBar.Suggestion] that the user clicked. This is `null` if the user is still
+ * interacting with the [AwesomeBar], or entered a search term or URL instead of clicking on a suggestion.
+ */
+data class AwesomeBarState(
```

Comment:
We could move `AwesomeBarState` into `feature-fxsuggest`... 🤔 I split it out like this to help with testing, but now realizing that it's also no longer tied directly to Fenix's `MetricsMiddleware`, and so it can live in a different / better place.

Reply:
If you're going with the direction above, moving the reducer there, then lets move the `AwesomeBarState` there too and rename it appropriately to be clear that this state and reducer are for FxSuggest only.

Might need to sleep on this and come back to you later.. I'm not sure about it yet.